### ISSUE TITLE
Possible incorrect plugin logging call caused by Logstash 5-rc1 switch to log4j2

### ISSUE THREAD

**Issue by martinbaillie** [logstash-plugins/logstash-output-redis#43]
Originally raised against Logstash. Please see https://github.com/elastic/logstash/issues/6048 for their input.
---
Hi folks,

I swapped in rc1 versions of ElasticSearch, Kibana, Logstash today in my Docker-based ELK stack. I hit the following issue with the Redis output plugin. Downgrading exclusively Logstash to previous alpha based builds (keeping E and K as rc1) works. Has something gone wrong with this output? It's late and I haven't done much research but couldn't see any obvious breaking changes in the documentation. The following example is a simple single line webserver access log entry.

Version: 5.0.0-rc1
Operating System: Linux (official upstream Logstash Docker image)
Config: -w "2" -b "1000"
```
outputs: 
redis {
host => "redis.host"
port => "6379"
data_type => "list"
key => "logstash"
}
```

Steps to Reproduce:
```
[2016-10-13T18:21:38,098][INFO ][logstash.inputs.udp      ] Starting UDP listener {:address=>"0.0.0.0:5000"}
[2016-10-13T18:21:38,098][INFO ][logstash.inputs.tcp      ] Starting tcp input listener {:address=>"0.0.0.0:5001"}
[2016-10-13T18:21:38,700][INFO ][logstash.pipeline        ] Starting pipeline {"id"=>"main", "pipeline.workers"=>2, "pipeline.batch.size"=>1000, "pipeline.batch.delay"=>5, "pipeline.max_inflight"=>2000}
[2016-10-13T18:21:38,701][INFO ][logstash.pipeline        ] Pipeline main started
[2016-10-13T18:21:38,855][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
[2016-10-13T18:21:39,338][WARN ][logstash.outputs.redis   ] Failed to send event to Redis {:event=>2016-10-13T07:21:38.613Z environment-logging_logstash-collector_1 - - [13/Oct/2016:13:15:10 +1100] 0.002 GET /some/kind/of/file.css 200 153 - 123.123.123.123 "http://hostname/something" -, :identity=>"default", :exception=>#<NameError: no method 'debug' for arguments (org.jruby.RubyHash,org.jruby.RubyHash) on Java::OrgApacheLoggingLog4jCore::Logger
  available overloads:
    (java.lang.String,org.apache.logging.log4j.util.Supplier[])
    (org.apache.logging.log4j.Marker,java.lang.String,java.lang.Object[])
    (org.apache.logging.log4j.Marker,java.lang.String,org.apache.logging.log4j.util.Supplier[])
    (java.lang.String,java.lang.Object[])>, :backtrace=>["/usr/share/logstash/logstash-core/lib/logstash/logging/logger.rb:41:in `debug'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-redis-3.0.2/lib/logstash/outputs/redis.rb:216:in `connect'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-redis-3.0.2/lib/logstash/outputs/redis.rb:241:in `send_to_redis'", "org/jruby/RubyProc.java:281:in `call'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-codec-json-3.0.2/lib/logstash/codecs/json.rb:42:in `encode'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-redis-3.0.2/lib/logstash/outputs/redis.rb:150:in `receive'", "/usr/share/logstash/logstash-core/lib/logstash/outputs/base.rb:92:in `multi_receive'", "org/jruby/RubyArray.java:1613:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/outputs/base.rb:92:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/logstash/output_delegator_strategies/legacy.rb:19:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/logstash/output_delegator.rb:42:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:302:in `output_batch'", "org/jruby/RubyHash.java:1342:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:301:in `output_batch'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:252:in `worker_loop'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:225:in `start_workers'"]}
```

**Comment by numbnut** [comment]
I'm facing the same issue and want to help reproducing it. I have set up a small docker-compose file to reproduce the bug easily. As a prerequisite you need docker and docker-compose installed on your system. Just create a file called docker-compose.yml with the following content.
```
version: '2'
services:
  logstash:
    image: logstash:5.0.0-rc1
    command: bash -c "ln -sf /dev/stdout /var/log/logstash/logstash-plain.log; logstash -e 'input { http { } } output { redis { host => \"redis\" data_type => \"list\" key => \"test\" } }'"
    ports:
      - "8080:8080"
    depends_on:
      - redis
  redis:
    image: redis:3.2
```
Startup the two containers in the foreground by invoking `docker-compose up`. In a different shell write some JSON to Logstash via HTTP.
```
curl -H "content-type: application/json" -XPUT 'http://localhost:8080/twitter/tweet/1' -d '{
    "user" : "kimchy",
    "post_date" : "2009-11-15T14:12:12",
    "message" : "trying out Elasticsearch"
}'
```
Now you should see a continuous error output to stdout in the first shell, which should look like this.
```
logstash_1  | [2016-10-19T16:16:45,326][WARN ][logstash.outputs.redis   ] Failed to send event to Redis {:event=>2016-10-19T16:16:43.499Z 172.18.0.1 trying out Elasticsearch, :identity=>"default", :exception=>#<NameError: no method 'debug' for arguments (org.jruby.RubyHash,org.jruby.RubyHash) on Java::OrgApacheLoggingLog4jCore::Logger
logstash_1  |   available overloads:
logstash_1  |     (org.apache.logging.log4j.Marker,java.lang.String,org.apache.logging.log4j.util.Supplier[])
logstash_1  |     (java.lang.String,java.lang.Object[])
logstash_1  |     (org.apache.logging.log4j.Marker,java.lang.String,java.lang.Object[])
logstash_1  |     (java.lang.String,org.apache.logging.log4j.util.Supplier[])>, :backtrace=>["/usr/share/logstash/logstash-core/lib/logstash/logging/logger.rb:41:in `debug'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-redis-3.0.2/lib/logstash/outputs/redis.rb:216:in `connect'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-redis-3.0.2/lib/logstash/outputs/redis.rb:241:in `send_to_redis'", "org/jruby/RubyProc.java:281:in `call'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-codec-json-3.0.2/lib/logstash/codecs/json.rb:42:in `encode'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-redis-3.0.2/lib/logstash/outputs/redis.rb:150:in `receive'", "/usr/share/logstash/logstash-core/lib/logstash/outputs/base.rb:92:in `multi_receive'", "org/jruby/RubyArray.java:1613:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/outputs/base.rb:92:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/logstash/output_delegator_strategies/legacy.rb:19:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/logstash/output_delegator.rb:42:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:302:in `output_batch'", "org/jruby/RubyHash.java:1342:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:301:in `output_batch'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:252:in `worker_loop'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:225:in `start_workers'"]}
```
Hope this helps to fix it soon, because the redis output plugin is useless for now. Thanks!

**Comment by IrlJidel** [comment]
Think this was fixed in 3.0.3 of plugin see 24635f635dce97257b2c1e3484275aacbd024ecc
